### PR TITLE
Hystrix configuration and hystrix integration tests implemented. Travis integration added.

### PR DESCRIPTION
**Pull request by yyenigun**

### CODE REVIEW

File: ozonosfer-spring/src/test/java/com/iyzico/ozonosfer/domain/RedisBasicRateLimiterHystrixIT.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        assertThat(circuitBreaker.allowRequest()).isFalse();
+    }
+
+    private void waitUntilCircuitBreakerOpens() throws InterruptedException {
+        /* one second is almost sufficient
+           borrowed from https://github.com/Netflix/Hystrix/blob/v1.5.5/hystrix-core/src/test/java/com/netflix/hystrix/HystrixCircuitBreakerTest.java#L140
+         */
+        Thread.sleep(1000);
+    }
+
```
Comment (burakahmetyoruk) [inline ozonosfer-spring/src/test/java/com/iyzico/ozonosfer/domain/RedisBasicRateLimiterHystrixIT.java]: 💃

File: ozonosfer-spring/src/test/java/com/iyzico/ozonosfer/domain/RedisBasicRateLimiterHystrixIT.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import com.netflix.hystrix.HystrixCommandKey;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.DirtiesContext;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
```
Comment (burakahmetyoruk) [inline ozonosfer-spring/src/test/java/com/iyzico/ozonosfer/domain/RedisBasicRateLimiterHystrixIT.java]: We can change with this one : org.assertj.core.api.Assertions.assertThat

File: ozonosfer-spring/src/test/java/com/iyzico/ozonosfer/domain/RedisBasicRateLimiterHystrixIT.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        // demonstrates circuit is actually closed
+        assertThat(circuitBreaker.isOpen()).isFalse();
+        assertThat(circuitBreaker.allowRequest()).isTrue();
+
+        // when redis is down
+        embeddedRedis.stopRedis();
+        Integer result = myLimitedService.rateLimitedMethod(request);
+
+        // then circuit is open and rate limited service executed!
+        waitUntilCircuitBreakerOpens();
```
Comment (ramazangirgin) [inline ozonosfer-spring/src/test/java/com/iyzico/ozonosfer/domain/RedisBasicRateLimiterHystrixIT.java]: https://github.com/awaitility/awaitility could be used for testing if circuitBreaker is opened in expected time range.

File: ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/Application.java [hunk 4]
```diff
@@ -1,9 +1,12 @@
...
 
+import com.iyzico.ozonosfer.infrastructure.configuration.OzonosferConfiguration;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.cloud.netflix.hystrix.EnableHystrix;
+import org.springframework.context.annotation.Import;
 
 @EnableHystrix
 @SpringBootApplication
+@Import(OzonosferConfiguration.class)
 public class Application {
```
Comment (ramazangirgin) [inline ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/Application.java]: Application could be moved to seperated module as example project or api side could be moved to seperated module. In this structure, if forces to use existing application.

File: ozonosfer-spring/src/test/java/com/iyzico/ozonosfer/domain/RedisBasicRateLimiterHystrixIT.java (hunk 3 above)
Comment (yyenigun) [reply ozonosfer-spring/src/test/java/com/iyzico/ozonosfer/domain/RedisBasicRateLimiterHystrixIT.java]: I will configure Docker and implement different integration tests with python.

File: ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/Application.java (hunk 4 above)
Comment (yyenigun) [reply ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/Application.java]: I will move it under tests folder.

**Pull request merged**